plugins/Other.js

```diff
@@ -804,6 +891,234 @@ async function other(fastify) {
+    aim.addMarkupEntity('TwoDimensionPoint', shapeIndex, points, imageReferenceUid, 1);
+  });
+
+  fastify.decorate('addLineToAim', (aim, line, shapeIndex, imageReferenceUid) => {
+    const { points } = line;
+    const markupId = aim.addMarkupEntity(
+      'TwoDimensionMultiPoint',
+      shapeIndex,
+      points,
+      imageReferenceUid,
+      1
+    );
+
+    fastify.createCalcEntity(aim, markupId, line);
+  });
+
+  fastify.decorate('createPointsArrForOsirix', osirixCoordinates => {
+    // unit, mean, stdDev, min, max, points, modality
+    const arr = osirixCoordinates.map(el => {
+      const coors = el.split(',');
+      const x = coors[0].trim().substr(1);
+      let y = coors[1].trim();
+      y = y.substr(0, y.length - 1);
+      return { x: parseFloat(x), y: parseFloat(y) };
+    });
+    return arr;
+  });
+  fastify.decorate('parseOsirix', docPath => {
+    return new Promise((resolve, reject) => {
+      try {
+        const osirixObj = plist.parse(fs.readFileSync(docPath, 'utf8'));
+        resolve(osirixObj);
+      } catch (err) {
+        reject(err);
+      }
+    });
+  });
+
+  fastify.decorate('getImageMetaDataforOsirix', (annotation, username) => {
+    return new Promise(async (resolve, reject) => {
+      try {
+        const { SOPInstanceUID, SeriesInstanceUID, StudyInstanceUID } = annotation.rois[0];
+        const parameters = {
+          instance: SOPInstanceUID,
+          series: SeriesInstanceUID,
+          study: StudyInstanceUID,
+        };
+        const seedData = await fastify.getImageMetadata(parameters);
+        const answers = fastify.getTemplateAnswers(seedData, annotation.name, '');
+        const merged = { ...seedData.aim, ...answers };
+        seedData.aim = merged;
+        seedData.user = { loginName: username, name: username };
+        resolve(seedData);
+      } catch (err) {
+        reject(new InternalError(`Getting data from image`, err));
+      }
+    });
+  });
+
+  // eslint-disable-next-line consistent-return
+  fastify.decorate('getTemplateAnswers', (metadata, annotationName, tempModality) => {
+    if (metadata.series) {
+      const { number, description, instanceNumber } = metadata.series;
+      const seriesModality = metadata.series.modality;
+      const comment = {
+        value: `${seriesModality} / ${description} / ${instanceNumber} / ${number}`,
+      };
+      // TODO: handle modality and suv hu
+      const modality = { value: tempModality };
+      const name = { value: annotationName };
+      const typeCode = [
+        {
+          code: 'ROI',
+          codeSystemName: '99EPAD',
+          'iso:displayName': { 'xmlns:iso': 'uri:iso.org:21090', value: 'ROI Only' },
+        },
+      ];
+      return { comment, modality, name, typeCode };
+    }
+  });
+
+  fastify.decorate('filterOsirixAnnotations', imagesArr => {
+    const filteredOsirix = {};
+    const nonSupported = [];
+    const supportedTypes = [19, 5, 14, 15, 10, 9, 28, 20, 11, 6];
```

Comment:
write comment saying if new supported objects are added createAimMarkups should be updated

Reply:
Done!